qlib/kernel/threadmgr/task_clone.rs

```diff
@@ -647,48 +647,55 @@ impl Task {
 
         return Ok(());
     }
-}
-
-pub fn CreateCloneTask(fromTask: &Task, toTask: &mut Task, userSp: u64) {
-    let mut from = fromTask.GetKernelSp();
-    let fromSp = fromTask.GetPtRegs() as *const _ as u64;
 
-    let mut to = toTask.GetKernelSp();
-    let toPtRegs = toTask.GetPtRegs();
+    pub fn copy_PtRegs_to_child(&self, clone_task: &mut Task, user_sp: u64) {
+        let parent_context = self.GetPtRegs() as *const _ as *const u64;
+        let child_context = clone_task.GetPtRegs() as *mut _ as *mut u64;
+        let context_size = core::mem::size_of::<PtRegs>();
 
-    unsafe {
-        while from >= fromSp {
-            *(to as *mut u64) = *(from as *const u64);
-            from -= 8;
-            to -= 8;
+        unsafe {
+            core::ptr::copy_nonoverlapping(parent_context, child_context, context_size);
         }
+        clone_task.GetPtRegs().set_stack_pointer(user_sp);
+    }
 
-        toTask.SetReady(1);
-        toTask.context.set_tls(fromTask.context.get_tls());
-        // TODO, what is this?
-        toTask.context.set_sp(toTask.GetPtRegs() as *const _ as u64 - 8);
-        toTask.context.set_para(userSp);
-        toTask.savefpsate = true;
-        toTask.archfpstate = Some(Box::new(
-            fromTask.archfpstate.as_ref().unwrap().Fork(),
-        ));
+    pub fn prepare_clone_entry(&mut self, entry_argument: u64) {
+        let entry_address = child_clone as u64;
+        let pid = 0;
+
+        self.context.set_para(entry_argument);
+        //
         // 1. set sys_clone return value to 0 to indicate child.
         // 2. set child pc (return addr of current call frame)
         //      to child_clone function
         #[cfg(target_arch = "x86_64")]
         {
-            toPtRegs.rax = 0;
-            toTask.context.place_on_stack(child_clone as u64);
```

Comment:
Maybe it's a hassle, but I suggest we don't modify x86 code on the arm feature branch.